File: Array/80. Remove Duplicates from Sorted Array II.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int length = 0;
        for(int i=0; i<nums.size(); i++)
        {
            if(length <2 || nums[length-2] != nums[i])
            {
                nums[length++] = nums[i];
            }
        }
        return length;
    }
};

File: Array/80. Remove Duplicates from Sorted Array II.py
"""
这道题一点也不容易

"""

class Solution:
    """
    @param: nums: An ineger array
    @return: An integer
    """
    def removeDuplicates(self, nums):
        # write your code here
        res = 0
        for n in nums:
            if res < 2 or n > nums[res-2]:
                nums[res] = n
                res += 1
        return res
            
